x/incentive/keeper/rewards_supply_test.go

```diff
@@ -680,13 +690,15 @@ func (suite *SupplyRewardsTestSuite) TestSynchronizeHardSupplyReward() {
 			}
 
 			// 2. Construct the parameter change proposal to update HardSupplyRewardPeriods param
-			pubProposal := params.NewParameterChangeProposal(
+			pubProposal := proposaltypes.NewParameterChangeProposal(
 				"Update hard supply rewards", "Adds a new reward coin to the incentive module's hard supply rewards.",
-				[]params.ParamChange{
+				[]proposaltypes.ParamChange{
 					{
 						Subspace: types.ModuleName,                         // target incentive module
 						Key:      string(types.KeyHardSupplyRewardPeriods), // target hard supply rewards key
-						Value:    string(suite.app.Codec().MustMarshalJSON(currIncentiveHardSupplyRewardPeriods)),
+						// TODO: currIncentiveHardSupplyRewardPeriods does not implement proto message so cannot use
+						// proto marshaller
+						Value: string(suite.app.LegacyAmino().MustMarshalJSON(&currIncentiveHardSupplyRewardPeriods)),
```

Comment:
LegacyAminio is the right codec to use for the `Value` in param change proposals. Cosmos still uses amino to encode params.

Reply:
Good to know, thanks